packages/driver/src/util/serialization/log.ts

```diff
@@ -0,0 +1,396 @@
+ */
+const preprocessDomElement = (props: HTMLElement) => {
+  // hydrate values in HTML copy so when serialized they show up correctly in snapshot. This is important for input boxes with typing and other 'value' attributes
+  props.querySelectorAll('input').forEach((input: HTMLInputElement) => {
+    input.setAttribute('value', input.value)
+    if (input.checked) {
+      input.setAttribute('checked', `${input.checked}`)
+    }
+  })
+
+  // TODO: figure out reifying option selection
+  // props.querySelectorAll('option').forEach((option) => {
+  //   if (option.selected) {
+  //     option.setAttribute('selected', option.selected)
+  //   }
+  // })
+
+  const el: PreprocessedHTMLElement = {
+    value: (props as HTMLInputElement)?.value,
+    type: (props as HTMLInputElement)?.type,
+    id: props?.id,
+    tagName: props.tagName,
+    attributes: {},
+    innerHTML: props.innerHTML,
+    serializationKey: 'dom',
+  }
+
+  // get all attributes and classes off the element
+  props.getAttributeNames().forEach((attributeName) => {
+    el.attributes[attributeName] = props.getAttribute(attributeName) || ''
+  })
+
+  return el
+}
+
+/**
+ * Takes an HTMLElement that might be a <body> for a given snapshot or any other element, likely pertaining to log consoleProps,
+ * on the page that needs to be preprocessed for serialization. The method here is to do a very shallow serialization,
+ * by trying to make the HTML as stateful as possible before preprocessing.
+ *
+ * @param {PreprocessedHTMLElement} props - a preprocessed element that was fed through postMessage() that need to be reified in the primary.
+ * @returns {HTMLElement} a reified element that can be used in a snapshot body or consoleProps.
+ */
+const reifyDomElement = (props: any) => {
+  const reifiedEl = document.createElement(props.tagName)
+
+  if (props.value) {
+    reifiedEl.value = props.value
+  }
+
+  try {
+    if (props.type) {
+      reifiedEl.type = props.type
+    }
+  } catch (e) {
+    // swallow this. certain elements this is a read-only property on (such as <select>)
+  }
+
+  if (props.id) {
+    reifiedEl.id = props.id
+  }
+
+  reifiedEl.innerHTML = props.innerHTML
+
+  Object.keys(props.attributes).forEach((attribute) => {
+    reifiedEl.setAttribute(attribute, props.attributes[attribute])
+  })
+
+  return reifiedEl
+}
+
+/**
+ * Attempts to take a generic data structure that is log-like and preprocess them for serialization.  * This generic may be/contain properties that are either
+ *  a) unserializable entirely
+ *  b) unserializable natively but can be processed to a serializable form (DOM elements or Functions)
+ *  c) serializable
+ *
+ * DOM elements are preprocessed via some key properties
+ * (attributes, classes, ids, tagName, value) including their innerHTML. Before the innerHTML is captures, `input`s are traversed to set their stateful value
```

Comment:
```suggestion
 * (attributes, classes, ids, tagName, value) including their innerHTML. Before the innerHTML is captured, `inputs are traversed to set their stateful value
```

Reply:
updated in [d28f924](https://github.com/cypress-io/cypress/pull/20949/commits/d28f924ebfcb37785672d6afed2588ca0a01b8ee)